### PR TITLE
[Deprecate _QQ_ 3] Mark _QQ_ as deprecated and normalize language ini parsing

### PR DESCRIPTION
**Pull request by andrepereiradasilva**
### Summary of Changes

As discussed in https://github.com/joomla/joomla-cms/issues/13421

Deprecate `_QQ_` and normalize parsing of ini language files.
- creates a new method JLanguageHelper::parseIniFile and marks LanguagesHelper::parseFile (helper of com_languages) as deprecated.
- makes the necessary adaptations

### Testing Instructions

1. Apply patch

2. Check all language strings show as usual. Including the ones with `"_QQ_"`

3. Confirm language debug still works.
Enable debug language in global config and system debug plugin.manually make some parsing errors in the en-GB.ini file and reload the page

4. Confirm language overrrides still works.
For instance, load a language string that uses escaped quotes (ex: JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED) and make some changes in the override. Save it and load it again to check if ok.

5. Confirm quoting in server side and client side still works (ex: add to isis template index.php)
```php
$app->enqueueMessage('Server side Test 1 (click to proceed to Client side Test 3): ' . JText::_('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED'));
JText::script('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED');
$this->addScriptDeclaration("alert('Client side Test 1: ' + Joomla.JText._('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED'));");
$this->addScriptDeclaration("alert('Client side Test 2: " . JText::_('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED', true) . "');");
$this->addScriptDeclaration("window.addEventListener('click', function () { Joomla.renderMessages({'error': ['Client side Test 3: ' + Joomla.JText._('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED')]}) });");
```

6. Any other test you could remember

### Documentation Changes Required

None

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
@frankmayer please check

File: libraries/joomla/language/helper.php [hunk 1]
```diff
@@ -380,4 +380,49 @@ public static function getContentLanguages($checkPublished = true, $checkInstall
...
+	public static function parseIniFile($fileName, $debug = false)
+	{
+		// Check if file exists.
+		if (!file_exists($fileName))
+		{
+			return array();
+		}
+
+		// @deprecated __DEPLOY_VERSION__ Usage of "_QQ_" is deprecated. Use escaped double quotes (\") instead.
+		if (!defined('_QQ_'))
```
Comment (frankmayer) [inline libraries/joomla/language/helper.php]: Why not define the constant in the top of the file and save the if()?
 Show all checks

File: libraries/joomla/language/language.php [hunk 2]
```diff
@@ -861,6 +825,7 @@ protected function parse($filename)
 	 *
 	 * @since   3.6.3
 	 * @throws  InvalidArgumentException
+	 * @todo    Debugging an ini file should not be coupled with the language object
```
Comment (mbabker) [inline libraries/joomla/language/language.php]: The debug process isn't coupled.  Simply having the method in this class does not make it coupled, unless you're considering tracking the files with errors in a coupling to the class.  Keep in mind static helper methods cannot have mocked dependencies (unless they are injected as parameters in the method signature) and generally require global services to be mocked (hello JFactory, I hate you for this).

Statics in general should be reserved for stateless and independent functionality.  To me, every single static method with a call to JFactory or any other method which is producing a singleton instance or creating an object in a way that precludes testing that method in isolation should be refactored to either be encapsulated in a class object (where the dependency can be injected) or have that dependency pushed in through the method signature.

All of that to say, I don't see a valid reason for this TODO.  It seems like asking for code to be moved for the sake of moving it.

File: libraries/joomla/language/helper.php (hunk 1 above)
Comment (andrepereiradasilva) [inline libraries/joomla/language/helper.php]: yes, is an extra if, i did it here to better warn where is used and where it will be removed.

Comment (frankmayer) [inline libraries/joomla/language/helper.php]: Oh, OK, makes sense.

File: libraries/joomla/language/language.php [hunk 3]
```diff
@@ -790,21 +773,13 @@ protected function loadLanguage($filename, $extension = 'unknown')
 	{
```
Comment (frankmayer) [inline libraries/joomla/language/language.php]: $filename -> $fileName ? (since you've done that in other methods...)

Comment (andrepereiradasilva) [inline libraries/joomla/language/language.php]: i have done only in the one i messed, let's leave that for another PR?

Comment (andrepereiradasilva) [inline libraries/joomla/language/language.php]: ok ok i see i changed this one too. will change

**Comment by frankmayer** [comment]
I have tested this item :white_check_mark: successfully on 71ea180034525484da888079b1782f2cdbbab4c8<br /><br />Tested according to instructions: OK
Code review: OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13451">issues.joomla.org/tracker/joomla-cms/13451</a>.</sub>

**Comment by infograf768** [comment]
Conflicts

**Comment by andrepereiradasilva** [comment]
conflicts fixed

**Comment by brianteeman** [comment]
I have tried locally to resolve the merge conflicts on this PR but I get
`Error: Class 'Joomla\CMS\Language\JLanguageHelper' not found: Class 'Joomla\CMS\Language\JLanguageHelper' not found`

I am guessing it is something to do with 
> creates a new method JLanguageHelper::parseIniFile and marks LanguagesHelper::parseFile (helper of com_languages) as deprecated.

Any clues?

**Comment by mbabker** [comment]
Any calls to `JLanguageHelper` in the `libraries/src` directory should either be prefixed with a leading slash (i.e. `\JLanguageHelper`) or use the full namespaced class name (i.e. `Joomla\CMS\Language\LanguageHelper`).

**Comment by brianteeman** [comment]
Thanks. I will check it in the morning

**Comment by brianteeman** [comment]
Closed see #20321
